GNUmakefile

```diff
@@ -240,7 +235,14 @@ test-internal:
 	@# hide it from travis as it exceeds their log limits and causes job to be
 	@# terminated (over 4MB and over 10k lines in the UI). We need to output
 	@# _something_ to stop them terminating us due to inactivity...
-	{ go test -v $(GOTEST_FLAGS) -tags '$(GOTAGS)' $(GOTEST_PKGS) 2>&1 ; echo $$? > exit-code ; } | tee test.log | egrep '^(ok|FAIL|panic:|--- FAIL|--- PASS)'
+	@echo "===================== submodule: sdk =====================" | tee -a test.log
+	cd sdk && { go test -v $(GOTEST_FLAGS) -tags '$(GOTAGS)' ./... 2>&1 ; echo $$? >> ../exit-code ; } | tee -a ../test.log | egrep '^(ok|FAIL|panic:|--- FAIL|--- PASS)'
+	@echo "===================== submodule: api =====================" | tee -a test.log
+	cd api && { go test -v $(GOTEST_FLAGS) -tags '$(GOTAGS)' ./... 2>&1 ; echo $$? >> ../exit-code ; } | tee -a ../test.log | egrep '^(ok|FAIL|panic:|--- FAIL|--- PASS)'
+	@echo "===================== submodule: root =====================" | tee -a test.log
+	{           go test -v $(GOTEST_FLAGS) -tags '$(GOTAGS)' ./... 2>&1 ; echo $$? >> exit-code    ; } | tee -a test.log    | egrep '^(ok|FAIL|panic:|--- FAIL|--- PASS)'
+	@# if everything worked fine then all 3 zeroes will be collapsed to a single zero here
+	@exit_codes="$$(sort -u exit-code)" ; echo "$$exit_codes" > exit-code
```

Comment:
This is a bit of a clever hack, but really this existing `test-internal` target is not lacking on cleverness.

Reply:
If this is 0,1,0 or 0,0,1 this would collapse to `01` (unique). Is that useful when displaying the exit code in the next line?

Reply:
I expanded the few lines after this that GitHub was hiding. I see that any non-0 we will just exit 1 anyway so it seems fine as is. I thought the exit code and reference to which module it came from might help people more.

Reply:
🤦‍♂ this is in a makefile where each line is a shell wow. sorry!